src/gemdat/path.py

```diff
@@ -34,6 +58,29 @@ def wrap(self, F: np.ndarray):
         X, Y, Z = F.shape
         self.sites = [(x % X, y % Y, z % Z) for x, y, z in self.sites]
 
+    def path_over_structure(self, structure: Structure, structure_map: dict):
+        """Find the nearest site of the structure to the path sites.
+
+        Parameters
+        ----------
+        structure : Structure
+            Reference structure
+        structure_map : dict
+            Dictionary of the closest site of the structure to each coordinate point
+        """
+        if not self.cart_sites:
+            raise ValueError(
+                'Cartesian coordinates of the path sites are not defined')
+
+        self.nearest_structure_label = [
+            structure.labels[structure_map.get(site, None)]
+            for site in self.cart_sites
+        ]
+        self.nearest_structure_coord = [
+            structure.cart_coords[structure_map.get(site, None)]
+            for site in self.cart_sites
+        ]
+
```

Comment:
I have refactored these operation to use a single loop, so I would keep them together.
The `path_over_structure()` method is required to plot the labels in `energy_along_path()`, but since I wanted to leave the label optiona, this is not explicitely called.